ci/Jenkinsfile.groovy

```diff
@@ -142,10 +143,20 @@ ansiColor('xterm') {
                             DT_RELEASE = 'True'
                         }
                         else if (env.BRANCH_NAME == 'master') {
-                            DT_BUILD_NUMBER = env.BUILD_ID
+                            DT_BUILD_NUMBER = sh(script: 'git rev-list --count master', returnStdout: true)
                         }
                         else {
-                            DT_BUILD_SUFFIX = env.BRANCH_NAME.replaceAll('[^\\w]+', '') + "." + env.BUILD_ID
+                            dir("/home/jenkins/repos/datatable") {
+                                sh "git checkout master"
+                                sh "git pull"
+                                sh "git checkout ${env.CHANGE_BRANCH}"
+                                def BRANCH_BUILD_ID = sh(script:
+                                  'git rev-list --count master..',
+                                  returnStdout: true
+                                )
+                                sh "git checkout master"
+                                DT_BUILD_SUFFIX = env.BRANCH_NAME.replaceAll('[^\\w]+', '') + "." + BRANCH_BUILD_ID
+                            }
```

Comment:
I'm still not convinced using a shared repository is a good idea. Consider the following scenario: branch A is forked from master, there are few commits there, and then it is converted into a PR. All is good:
```
      /---*---*---* (A)
-----o  (master)
```
Now, it turns out we need to make some fixes in the RP, so a new commit is pushed into branch A. However, in the meanwhile, some other PR was merged to master, and the tree now looks like this:
```
      /---*---*---*---* (A)
-----o----o  (master)
```
There is no longer a direct path between A and master, and the rev-list-count command will fail (I think)

Reply:
Or another scenario: 2 separate PRs attempt to run their tests at the same time, and consequently they run `git pull` on the same local repository at the same time. Since git was not designed for a parallel local access, a data corruption may occur.

Reply:
I'm also not sure this is a good idea. May only work if we have 1 PR running on 1 machine at the same time, and if no one touches the `/home/jenkins/repos/datatable` folder meanwhile (I don't think we have a guarantee for that). For the moment I want to see if the new PR build id suffix will work as expected on both AppVeyor and Jenkins, and it seems it does work. Now we just need to make it safe.

>There is no longer a direct path between A and master, and the rev-list-count command will fail (I think)

This is exactly the situation we have now, and I don't think anything failed because of the fact https://github.com/h2oai/datatable/pull/2445 was merged.

Reply:
Well, there is actually a problem for the last commit Jenkins has `18` as the PR build suffux and AppVeyor has `19` (it seems that AppVeyor is wrong...).